### PR TITLE
Add ROS Noetic CI infrastructure

### PR DESCRIPTION
**Pull request by hidmic**
### Proposed changes

Follow-up PR after #209. This patch adds CI infrastructure to build `beluga` (and only `beluga` for now) against ROS Noetic in Ubuntu Focal.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: tools/run-clang-tidy.sh [hunk 1]
```diff
@@ -20,7 +20,11 @@
 
 set -o errexit -o xtrace
 
-ROS_PACKAGES="beluga beluga_system_tests"
+if [ "${ROS_DISTRO}" != "noetic" ]; then
+    ROS_PACKAGES="beluga beluga_system_tests"
+else
+    ROS_PACKAGES="beluga"
```
Comment (nahueespinosa) [inline tools/run-clang-tidy.sh:26]: Regarding the CI static analysis failure, we can probably add `-clang-diagnostic-gnu-zero-variadic-macro-arguments` to the checks in `.clang-tidy`.

The include order warning is weird, but adding a newline in between in `test_strongly_typed_numeric.cpp` may fix it:
```cpp
#include <gtest/gtest.h>

#include <beluga/type_traits/strongly_typed_numeric.hpp>
```

**Review by nahueespinosa — APPROVED** [review]
LGTM!

File: .github/workflows/ci_pipeline.yml [hunk 2]
```diff
@@ -65,20 +97,13 @@ jobs:
           fail_ci_if_error: true
           verbose: true
```
Comment (nahueespinosa) [inline .github/workflows/ci_pipeline.yml:98]: Should we add a condition to upload documentation artifacts and code coverage reports in only one of the jobs?

Something like:
```yaml
- if: ${{ matrix.ros_distro == 'humble' }}
  name: Upload documentation artifacts
```

File: tools/run-clang-tidy.sh (hunk 1 above)
Comment (nahueespinosa) [reply tools/run-clang-tidy.sh:26]: Hmm, there is also an unused parameter in a googletest macro... I wonder if it is related to the googletest version or the differences between `clang-tidy-10` and `clang-tidy-14`.

File: .github/workflows/ci_pipeline.yml (hunk 2 above)
Comment (nahueespinosa) [reply .github/workflows/ci_pipeline.yml:98]: Or better yet, do something like what is proposed in https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-expanding-configurations.

```yaml
strategy:
      fail-fast: false
      matrix:
        ros_distro:
          - noetic
          - humble
      include:
          - ros_distro: humble
            upload_artifacts: true
```

```yaml
- if: ${{ matrix.upload_artifacts }}
  name: Upload documentation artifacts
  
- if: ${{ matrix.upload_artifacts }}
  name: Upload code coverage report
```

File: tools/run-clang-tidy.sh (hunk 1 above)
Comment (hidmic) [reply tools/run-clang-tidy.sh:26]: I believe it's https://github.com/google/googletest/issues/2650, which has been fixed for googletest 1.11 but not backported to 1.10. I tried and failed to handle googletest versions in CMake -- Kitware's GTest module doesn't detect any versions. Dropping the warning seems like the only path forward.

Comment (hidmic) [reply tools/run-clang-tidy.sh:26]: >  adding a newline in-between in test_strongly_typed_numeric.cpp may fix it:

Indeed.

File: .github/workflows/ci_pipeline.yml (hunk 2 above)
Comment (hidmic) [reply .github/workflows/ci_pipeline.yml:98]: OK with uploading documentation artifacts once, but it does make sense to have coverage reports in Noetic as well. Not sure how to configure `codecov` for that though.

File: tools/run-clang-tidy.sh (hunk 1 above)
Comment (hidmic) [reply tools/run-clang-tidy.sh:26]: > we can probably add -clang-diagnostic-gnu-zero-variadic-macro-arguments to the checks in .clang-tidy.

This doesn't seem to work. Passing `-Wno-gnu-zero-variadic-macro-arguments` to the compiler does (and thus to the compiler_commands.json file) does.

**Review by nahueespinosa — APPROVED** [review]

**Pull request merged**